Formula/byobu.rb

```diff
@@ -36,13 +36,23 @@ def install
     end
     system "./configure", "--prefix=#{prefix}"
     system "make", "install"
-  end
 
-  def caveats
-    <<~EOS
-      Add the following to your shell configuration file:
-        export BYOBU_PREFIX=#{HOMEBREW_PREFIX}
-    EOS
+    python_binaries_without_extensions = Dir.chdir(lib/"byobu/include/") do
+      Dir.glob("*.py").map { |path| File.basename(path, ".py") }
+    end
+
+    first_newt_python_dep = Formula["newt"].deps.find { |d| d.name.match?(/^python@\d\.\d+$/) }
+
+    # This raises NoMethodError (undefined method `name' for nil:NilClass) if
+    # newt unexpectedly doesn't have any Python dependencies.
+    byobu_python = Formula[first_newt_python_dep.name].opt_bin/"python3"
```

Comment:
I didn't know `to_formula` is allowed for this repository as well.

> This method is part of a private API. This method may only be used in the [Homebrew/brew](https://github.com/Homebrew/brew) repository. Third parties should avoid using this method if possible, as it may be removed or changed without warning.
https://rubydoc.brew.sh/Dependency.html#to_formula-instance_method